clang/include/clang/Basic/Builtins.td

```diff
@@ -4883,6 +4882,12 @@ def HLSLRadians : LangBuiltin<"HLSL_LANG"> {
   let Prototype = "void(...)";
 }
 
+def HLSLBufferUpdateCounter : LangBuiltin<"HLSL_LANG"> {
+  let Spellings = ["__builtin_hlsl_buffer_update_counter"];
+  let Attributes = [NoThrow, Const];
```

Comment:
I don't know what effect these attributes have or if I'm understanding Const correctly, but if that implies it doesn't change any memory, it does.